shop/src/main/java/ro/msg/learning/shop/Service/Implementation/ProductService.java

```diff
@@ -0,0 +1,111 @@
+package ro.msg.learning.shop.Service.Implementation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.DTO.ProductAndProductCategoryDto;
+import ro.msg.learning.shop.Domain.Location;
+import ro.msg.learning.shop.Domain.Product;
+import ro.msg.learning.shop.Domain.ProductCategory;
+import ro.msg.learning.shop.Domain.Stock;
+import ro.msg.learning.shop.Exception.*;
+import ro.msg.learning.shop.Mapper.ProductAndProductCategoryTranslatorMapper;
+import ro.msg.learning.shop.Repository.ILocationRepository;
+import ro.msg.learning.shop.Repository.IProductCategoryRepository;
+import ro.msg.learning.shop.Repository.IProductRepository;
+import ro.msg.learning.shop.Repository.IStockRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class ProductService {
+    @Autowired
```

Comment:
So there are many ways to do this DI in Spring and `@Autowired` is one of them. However there seems to be a better approach that we can use in order to achieve the same thing, but maybe more readable, testable and reliable.
This is nice to be implemented in projects that use lombok as we do.
Basically instead of using `@Autowired` we need to make sure that we have a constructor with all fields/arguments, and since this is the only constructor, Spring will inject the dependencies through it (the fields must be final).
References: 
1. https://stackoverflow.com/questions/68538851/lombok-and-autowired
2. https://betulsahinn.medium.com/why-is-autowired-annotation-not-recommended-4939c46da1f8

Reply:
Do this for all things that need to be injected.